### PR TITLE
[IMPROVED] Fix flaky tests

### PR DESCRIPTION
**Pull request by piotrpio**
Signed-off-by: Piotr Piotrowski [piotr@synadia.com](mailto:piotr@synadia.com)

### CODE REVIEW

File: test/context_test.go [hunk 1]
```diff
@@ -220,20 +220,28 @@ func testContextRequestWithCancel(t *testing.T, nc *nats.Conn) {
...
+		t.Fatalf("Expected to be able to subscribe: %s", err)
+	}
+	defer sub1.Unsubscribe()
+	sub2, err := nc.Subscribe("slower", func(m *nats.Msg) {
 		// we know this request will take longer so extend the timeout
 		expirationTimer.Reset(100 * time.Millisecond)
 
 		// slower reply which would have hit original timeout
-		time.Sleep(90 * time.Millisecond)
+		time.Sleep(70 * time.Millisecond)
```
Comment (Jarema) [inline test/context_test.go:237]: What's the reason for this change here?

File: test/js_test.go [hunk 2]
```diff
@@ -593,17 +593,22 @@ func TestJetStreamSubscribe(t *testing.T) {
...
-	if info.Delivered.Consumer != uint64(toSend) {
-		t.Fatalf("Expected to have received all %d messages, got %d", toSend, info.Delivered.Consumer)
-	}
-	// Make sure we auto-ack'd
-	if info.AckFloor.Consumer != uint64(toSend) {
-		t.Fatalf("Expected to have ack'd all %d messages, got ack floor of %d", toSend, info.AckFloor.Consumer)
-	}
+	// We may need to retry this check since the acks sent by the client have to be processed
+	// on the server.
+	checkFor(t, 2*time.Second, 100*time.Millisecond, func() error {
```
Comment (Jarema) [inline test/js_test.go:598]: For sake of slow CI runs, maybe it's better to increase the max Duration to 10 seconds?

File: jetstream/test/kv_test.go [hunk 3]
```diff
@@ -1213,6 +1213,35 @@ func TestListKeyValueStores(t *testing.T) {
 }
 
 func TestKeyValueMirrorCrossDomains(t *testing.T) {
+	keyExists := func(t *testing.T, kv jetstream.KeyValue, key string, expected string) jetstream.KeyValueEntry {
+		var e jetstream.KeyValueEntry
+		var err error
+		checkFor(t, 5*time.Second, 10*time.Millisecond, func() error {
```
Comment (Jarema) [inline jetstream/test/kv_test.go:1219]: If we will hit very slow/busy worker, maybe it's worth to have all max timeouts set to 10 seconds.

**Review by Jarema — CHANGES_REQUESTED** [review]
Nice improvement of the CI relaiblity. Howeve, I think it would be nice to increase to Durations to make it more durable against slow CI workers.

File: test/context_test.go (hunk 1 above)
Comment (piotrpio) [reply test/context_test.go:237]: This test depends on request either timing out or not depending on the circumstances. The expiration timer is reset in each subscribe callback to 100ms - sleeping for 90ms was making it fail quite often if there the subsequent request had 10ms delay. Now it's 30ms "buffer", which makes it less, but still, flaky...

I get why this test is useful but its tough to fix it entirely while retaining the logic.

Comment (Jarema) [reply test/context_test.go:237]: Oh, gotcha. That feels pretty flaky no matter what we set here, seeing faster or slower machines running it...
